neqo-transport/src/crypto.rs

```diff
@@ -1358,6 +1360,80 @@ pub enum CryptoStreams {
     },
 }
 
+/// [Fisher–Yates shuffle](https://en.wikipedia.org/wiki/Fisher%E2%80%93Yates_shuffle#Modern_method)
+/// Modified to make sure no element stays in place if `a` has more than one element.
+///
+/// TODO: We could use <https://docs.rs/rand/latest/rand/seq/trait.SliceRandom.html#tymethod.shuffle>
+/// instead, but I assume we want to only use the NSS RNG in neqo?
+fn shuffle<T>(a: &mut [T]) {
+    // To shuffle an array a of n elements (indices 0..n-1)
+    let n: usize = a.len();
+    if n < 2 {
```

Comment:
Given that each sampling draws from 0..256, it might be worth limiting the number of cuts we take.  I seriously doubt that we'll have to worry too much if we exceed 256 slices, but your current code has the potential to exceed that limit if a handshake message is >1k.  That seems like it might be a limit we hit often enough for this to be a problem.

I'm tempted to suggest that you increase the length of the LDH character span to compensate for that a little.  If you increase the number to 5, then you get to 1.5k.  That's still on the edge for me, but the chance that you'll get that in a typical ClientHello is small.

In addition, I think that you should limit the number of values that you return from `ascii_sequences()`.  For certificates, for example, the number of splits is going to be high just because the messages are long.  Limiting the number of cuts is going to work out fine, I think.

In terms of implementing that, I would suggest that rather than returning `Vec<(usize, usize)>` (or `Vec<Range<usize>>` as I suggest) from `ascii_sequences()` you instead return `impl Iterator<Item = Range<usize>>` and then you call `.take(256)` at the call site.

Reply:
I fixed the `256` bug.

The remaining question is whether we want to limit the number of chunks in some way.

Reply:
@martinthomson AFAIK the only thing left is to limit the number of generated chunks. I'm wondering if we should look for the _n_ largest ASCII sequences (which _should_ include the SNI) and then use their midpoints as split points? Any suggestion for a good _n_?

Reply:
Searching for the largest will involve unnecessarily expensive sorting (far better to parse the ClientHello in that case) or multiple passes.  I honestly think that n=32 for 4 or 5 byte sequences of LDH+. is more than enough.  (Run a test with ML-KEM enabled a few hundred times at 4 and if you hit the limit too often, go with 5.)

Reply:
Didn't see this until now. Let me know what you think of cb63418e54903517dbfb3b8c556cb70b21911099, which does what I outlined above. Happy to go a different way, if you prefer.